junit-platform-engine/src/main/java/org/junit/platform/engine/support/store/NamespacedHierarchicalStore.java

```diff
@@ -139,6 +150,8 @@ public <T> T get(N namespace, Object key, Class<T> requiredType) throws Namespac
 	 * @return the stored value; may be {@code null}
 	 */
 	public <K, V> Object getOrComputeIfAbsent(N namespace, K key, Function<K, V> defaultCreator) {
+		if (this.closed)
+			illegalQueryAfterClose();
```

Comment:
```suggestion
		if (this.closed) {
			rejectQueryAfterClose();
		}
```

Please use if-blocks here and in the rest of the modified methods.